### ISSUE TITLE
Reordered Grid of columns htrows exception when moving left with the keyboard

### ISSUE THREAD

**Issue by DGH2112** [JAM-Software/Virtual-TreeView#1247]
Hi,

I've been updating my application to the latest version of VT (V7.6.6, with RS11.3) and I've come across, what I believe, is an error in the code. I can reproduce this with the Advanced demo as follows:

* Open the demo;
* Select the grid example (5th button);
* Reorder the columns so that "Order Date" is before "Last Name";
* Select the last cell (Last Name) in the first row (the row is unimportant);
* Using the keyboard, press the left arrow key (selecting with the mouse does not throw the error).

The exception is thrown in the method `TVirtualTreeColumns.GetPreviousColumn()`. I believe the marked line below
is testing the column index and not the position.

```
function TVirtualTreeColumns.GetPreviousColumn(Column : TColumnIndex) : TColumnIndex;

var
  Position : Integer;

begin
  if Column < 0 then
    Result := InvalidColumn
  else
  begin
    Position := Items[Column].Position;
    if Position > 0 then
      Result := FPositionToIndex[Position - 1] // Testing index not position???
    else
      Result := InvalidColumn;
    Assert(Position <> Result, 'The previous column must not have the same position as the given column.');
  end;
end;
```

If you need any more information, please let me know.

regards
Dave.

**Comment by joachimmarder** [comment]
> I've been updating my application to the latest version of VT (V7.6.6, with RS11.3)

The latest version is V8.0.2, by the way, but here the problem is also present.

>  I believe the marked line below is testing the column index and not the position.

Indeed, this seems to be a copy and paste error.

**Comment by joachimmarder** [comment]
Fixed with commit 10f854d
